### PR TITLE
expose uploader validation via action

### PR DESCRIPTION
**Pull request by joshchernoff**
I'm not 100% sure if this is the correct way to expose the validate logic but I think its a start. 

The main point of why I want to expose this method is that I would like to build a changeset method to check the validation ie: `validate_attachments` that will be a macro that expands in via `Arc.Ecto.Schema`

### CODE REVIEW

File: lib/arc/actions/validate.ex [hunk 1]
```diff
@@ -0,0 +1,24 @@
+defmodule Arc.Actions.Validate do
+
+  defmacro __using__(_) do
+    quote do
+      def valid?(args), do: Arc.Actions.Validate.valid?(__MODULE__, args)
+    end
+  end
+
+  def valid?(definition, {file, scope}) when is_binary(file) or is_map(file) do
+    case Arc.File.new(file) do
```
Comment (stavro) [inline lib/arc/actions/validate.ex]: `Arc.File.new("https://some-remote-path")` will download the file currently.  So, as it currently stands anyone who tries to do validation will end up downloading the file twice (once during validation, and once during the storage action...

Hmm...

Comment (digitalcake) [inline lib/arc/actions/validate.ex]: I had not thought about that. Thanks for reviewing. I will think about a solution to this problem.

Comment (stavro) [inline lib/arc/actions/validate.ex]: Maybe the answer is to provide a different interface for downloading remote images (as some sort of pre-processing step that the user would do explicitly?)

I'm not really a fan of how it's hidden inside here.  But either way I'd like to think through that use case

**Comment by stavro** [comment]
Closing this for now. I am open to this idea, but as this PR currently stands I do want want to merge it.

Thank you!

**Pull request closed without merging**